### PR TITLE
Docker: Set RMQ consumer_timeout to undefined

### PR DESCRIPTION
**Pull request by unkcpz**
Fixes https://aiida.discourse.group/t/workchain-except/152/36

### CODE REVIEW

File: .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh [hunk 1]
```diff
@@ -14,8 +14,8 @@ echo MNESIA_BASE="${RABBITMQ_DATA_DIR}" >> "${RMQ_ETC_DIR}/rabbitmq-env.conf"
 echo LOG_BASE="${RABBITMQ_DATA_DIR}/log" >> "${RMQ_ETC_DIR}/rabbitmq-env.conf"
 
 # using workaround from https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use
-# set timeout to 100 hours
-echo "consumer_timeout=3600000" >> "${RMQ_ETC_DIR}/rabbitmq.conf"
+# 1000 hours in milliseconds
+echo "consumer_timeout=3600000000" >> "${RMQ_ETC_DIR}/rabbitmq.conf"
```
Comment (sphuber) [inline .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh:18]: Wouldn't it perhaps be better to add the `advanced.confgi` approach instead, which allows us to simply set the timeout to undefined? I think the following should work
```suggestion
cat > "${RMQ_ETC_DIR}/advanced.config" <<EOF
%% advanced.config
[
  {rabbit, [
    {consumer_timeout, undefined}
  ]}
].
EOF
```

**Review by sphuber — CHANGES_REQUESTED** [review]

File: .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh (hunk 1 above)
Comment (unkcpz) [reply .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh:18]: Ah, cool, I didn't know this. Will give it a build locally and test.

**Comment by unkcpz** [comment]
Hi @sphuber, in discourse you mentioned you can reproduce the issue correct? Can you try with `ghcr.io/aiidateam/aiida-core-with-services:pr-6189` from this PR and `ghcr.io/aiidateam/aiida-core-with-services:edge` to make sure this solves the problem?

File: .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh (hunk 1 above)
Comment (unkcpz) [reply .docker/aiida-core-with-services/s6-assets/init/rabbitmq-init.sh]: Changed and add test checking it is set, but better to run a calculation to test it really activated.

**Comment by sphuber** [comment]
@unkcpz I baked this recipe locally so that I can run a container and try and debug why the build on Github actions was failing. I managed to build and start a container, but only as the normal user. When I try to `-u root` the container hangs during startup because it cannot start the services as root for some reason. Is there a way to become root inside the container though? I want to check `rabbitmqctl environment` to make sure the `consumer_timeout` is properly set to `undefined`. Note that `verdi status` actually runs fine in my locally baked image. Don't understand why it fails on GHA.

**Comment by unkcpz** [comment]
Yes, I found the same that `-u root` will stuck the initial processes. To root, running a container first and run `docker exec -it -u 0 <container id> bash` to get into the running container as root.

**Comment by unkcpz** [comment]
The failed test is on arm64 build, I can test it from my mackbook.

**Comment by unkcpz** [comment]
I tested locally on my M1 Mac, and all passed. The failed test is the timeout of `_docker_service_wait`, I roll back to the old `time.sleep` method to make sure the profile is ready before the test. The downside is the test takes a bit longer. If this makes the test pass, I will not combine the conftest polish changes in this PR.

**Comment by sphuber** [comment]
> Yes, I found the same that `-u root` will stuck the initial processes. To root, running a container first and run `docker exec -it -u 0 <container id> bash` to get into the running container as root.

Excellent, that worked, thanks. I verified that `rabbitmqctl environment` returns
```
      {consumer_timeout,undefined},
```
so I am pretty sure it worked.

**Review by sphuber — APPROVED** [review]

**Pull request merged**